syft/workers/base.py

```diff
@@ -542,13 +542,12 @@ def execute_communication_action(self, action: CommunicationAction) -> PointerTe
         else:
             obj = self.get_obj(obj_id)
             response = source_worker.send(obj, *destinations, **kwargs_)
-
-            response = hook_args.register_response("send", response, [sy.ID_PROVIDER.pop()], self)
-
-            # @lariffle: We only remove remote objects when the operations are inplace
-            # otherwise we could have stale pointers which we really want to avoid.
-            # TODO: needs more discussion
-            if kwargs_.get("inplace"):
+            if kwargs_.get("requires_grad", False):
```

Comment:
I don't follow why `requires_grad` controls whether or not we register the response. What's the rationale for that? Can we make it clearer in this code?

Reply:
### If a tensor requires grad, it mustn't get destroyed till the `loss.backward()` is invoked. 
Basically, we shouldn't delete the tensor when `requires_grad` is `True`, but always when its `None` or `False`. So, for it to be preserved till backprop, the tensor has to be registered.

Reply:
Hmm. Well, maybe we could add a comment that conveys that part?

It would probably be clearer if the flag was called `backprop`, but that change would extend beyond the scope of this PR.

Reply:
Currently there is another PR in the happening that fixes some GC issues with the backprop and involves this part of the codebase. I will address your comment in them. 👍